sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -327,8 +327,8 @@ object IntervalUtils {
         toDTInterval(day, hour, minute, secondAndMicro(second, micro),
           finalSign(firstSign, secondSign))
 
-      case hourMinuteRegex(sign, hour, minute) =>
-        checkDTIntervalStringDataType(DT(DT.HOUR, DT.MINUTE))
+      case hourMinuteRegex(sign, hour, minute)
+        if startField == DT.HOUR && endField == DT.MINUTE =>
```

Comment:
How about to make all place like this? FYI @cloud-fan @MaxGekk

Reply:
SGTM